### PR TITLE
Mask XSAVE PKRU bit in C2/T3 templates

### PR DESCRIPTION
**Pull request by sandreim**
## Reason for This PR

Fixes an inconsistency in the templates - the OSPKE feature bit is masked, but the XSAVE PKRU bit is not. Snapshots created on a CPU with OSPKE from a T2/C3 uVM would fail to load on a CPU without OSPKE (Broadwell): https://elixir.bootlin.com/linux/latest/source/arch/x86/kvm/x86.c#L4299.

## Description of Changes

Provided in commit message.

- [ ] This functionality can be added in [`rust-vmm`](https://github.com/rust-vmm).

## License Acceptance

By submitting this pull request, I confirm that my contribution is made under
the terms of the Apache 2.0 license.

## PR Checklist

`[Author TODO: Meet these criteria.]`
`[Reviewer TODO: Verify that these criteria are met. Request changes if not]`

- [ ] All commits in this PR are signed (`git commit -s`).
- [ ] The reason for this PR is clearly provided (issue no. or explanation).
- [ ] The description of changes is clear and encompassing.
- [ ] Any required documentation changes (code and docs) are included in this PR.
- [ ] Any newly added `unsafe` code is properly documented.
- [ ] Any API changes are reflected in `firecracker/swagger.yaml`.
- [ ] Any user-facing changes are mentioned in `CHANGELOG.md`.
- [ ] All added/changed functionality is tested.

### CODE REVIEW

File: src/cpuid/src/cpu_leaf.rs [hunk 1]
```diff
@@ -240,6 +240,7 @@ pub mod leaf_0xd {
 
             pub const MPX_STATE_BITRANGE: BitRange = bit_range!(4, 3);
             pub const AVX512_STATE_BITRANGE: BitRange = bit_range!(7, 5);
+            pub const PKRU_SHIFT: u32 = 9;
```
Comment (serban300) [inline src/cpuid/src/cpu_leaf.rs:243]: Nit: I would rename this to `PKRU_BITINDEX` in order to stay consistent with the other variables.

**Comment by dianpopa** [comment]
It is not clear to me if we want to hide the feature also on hardware that has support for it?

**Review by andreeaflorescu — COMMENTED** [review]
Good catch on the inconsistency! You should also update the integration test with the newly added features in the template so that we also get a regression test: https://github.com/firecracker-microvm/firecracker/blob/master/tests/integration_tests/functional/test_cpu_features.py#L281

Just wanted to call out that for other features we might need to update the restore path to only set the features that are available on the cpu, as masking unavailable features on both templates would make the templates useless.

LGTM, will approve once you also update the integration test.

File: src/cpuid/src/cpu_leaf.rs (hunk 1 above)
Comment (sandreim) [reply src/cpuid/src/cpu_leaf.rs]: done

**Comment by sandreim** [comment]
> Good catch on the inconsistency! You should also update the integration test with the newly added features in the template so that we also get a regression test: https://github.com/firecracker-microvm/firecracker/blob/master/tests/integration_tests/functional/test_cpu_features.py#L281
> 
There is no new feature added, just disabling the XCR0 PKRU bit. I've updated the existing template integration tests to look out for the PKRU bit.

> Just wanted to call out that for other features we might need to update the restore path to only set the features that are available on the cpu, as masking unavailable features on both templates would make the templates useless.
> 
 What CPUID we expose to the guest must remain the same during the microvm lifecycle which includes snapshots. Changing the exposed features on restore might break workloads that were running/using them when the snapshot was created.

> LGTM, will approve once you also update the integration test.

Please take another look.

**Review by andreeaflorescu — APPROVED** [review]

**Review by serban300 — APPROVED** [review]

**Pull request merged**